Services/src/main/java/ev/projects/services/LogService.java

```diff
@@ -0,0 +1,58 @@
+package ev.projects.services;
+
+import ev.projects.models.Log;
+import ev.projects.repositories.ILogRepository;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class LogService implements ILogService {
+
+    private ILogRepository logRepository;
```

Comment:
I used to use field injection, however now my IDE says it's not recommended.

Reply:
I could use field injection of course, because it's a simulation. But what do you think about this post: https://stackoverflow.com/questions/39890849/what-exactly-is-field-injection-and-how-to-avoid-it?

Reply:
Have implemented a field injection in 1e34946.

Reply:
What IDE are you using which says it's not recommended?

Reply:
> I could use field injection of course, because it's a simulation. But what do you think about this post: https://stackoverflow.com/questions/39890849/what-exactly-is-field-injection-and-how-to-avoid-it?

Never in 8 years encountered any of drawbacks listed in the thread. Usually in a project there is a consensus. Which approach to use. When there is no good reason to use setters from my experience everyone use field injection by default.  Regarding tests - this post is from 2016,maybe mockito was not so advanced at the time, you are more likely to have issues with nested injections and private methods than autowiring on fields. But I guess anyone can have an opinion. Whatever is more attractive for you...